hypershift-operator/controllers/nodepool/kubevirt/kubevirt.go

```diff
@@ -238,6 +239,8 @@ func MachineTemplateSpec(image string, nodePool *hyperv1.NodePool) *capikubevirt
 	vmTemplate.Spec.Template.ObjectMeta.Labels[nodePoolNameLabelKey] = nodePool.Name
 	vmTemplate.ObjectMeta.Labels[nodePoolNameLabelKey] = nodePool.Name
 
+	vmTemplate.ObjectMeta.Namespace = manifests.HostedControlPlaneNamespace(nodePool.Namespace, nodePool.Name).Name
```

Comment:
fyi, the `nodepool.Name` doesn't necessarily correlate to the HostedCluster's name. that could result in some unexpected results here with how the namespace is generated.

Reply:
yes, i've noticed that.
i'm now passing the hcluster object to `MachineTemplateSpec()` for its name & namespace.